CMakeLists.txt

```
@@ -3,11 +3,23 @@ if(NOT CMAKE_VERSION VERSION_LESS 3.12)
   cmake_policy(SET CMP0074 NEW) # Don't complain about using BOOST_ROOT...
 endif()
 
-set(HIPSYCL_VERSION_MAJOR 23)
-set(HIPSYCL_VERSION_MINOR 10)
-set(HIPSYCL_VERSION_PATCH 0)
 
-project(hipSYCL VERSION ${HIPSYCL_VERSION_MAJOR}.${HIPSYCL_VERSION_MINOR}.${HIPSYCL_VERSION_PATCH})
+set(ACPP_VERSION_MAJOR 23)
+set(ACPP_VERSION_MINOR 10)
+set(ACPP_VERSION_PATCH 0)
+if(NOT ACPP_VERSION_SUFFIX)
+  # Get the latest abbreviated commit hash of the working branch
+  execute_process(
+    COMMAND git log -1 --format=%h
+    WORKING_DIRECTORY ${CMAKE_SOURCE_DIR}
+    OUTPUT_VARIABLE ACPP_GIT_COMMIT_HASH
+    OUTPUT_STRIP_TRAILING_WHITESPACE
+  )
+
+  set(ACPP_VERSION_SUFFIX "+git${ACPP_GIT_COMMIT_HASH}")
```

When building from a tarball (or if git fails for any other reason), the `ACPP_GIT_COMMIT_HASH` will be just `+git`, which is quite misleading. I suggest checking the success of the git query:

```suggestion
    OUTPUT_VARIABLE ACPP_GIT_COMMIT_HASH
    RESULT_VARIABLE GIT_RETURN_CODE
    OUTPUT_STRIP_TRAILING_WHITESPACE
  )

  if (GIT_RETURN_CODE EQUAL 0)
    set(ACPP_VERSION_SUFFIX "+git${ACPP_GIT_COMMIT_HASH}")
  endif()
```

---

Good point, thanks!